core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -127,6 +130,11 @@ private[spark] object SparkUI {
   val DEFAULT_RETAINED_STAGES = 1000
   val DEFAULT_RETAINED_JOBS = 1000
 
+  val listenerRegisters: Iterable[SparkListenerRegister] = {
```

Comment:
Do you need to keep this in a `val`? Could it be a local variable in `createHistoryUI`?